packages/form-js-viewer/src/render/components/form-fields/parts/Datepicker.js

```diff
@@ -0,0 +1,183 @@
+  const [ isInputDirty, setIsInputDirty ] = useState(false);
+  const [ forceRefreshValue, setForceRefreshValue ] = useState(false);
+
+  const enterKeyboardEvent = useMemo(() => new KeyboardEvent('keydown', {
+    code: 'Enter',
+    key: 'Enter',
+    charCode: 13,
+    keyCode: 13,
+    view: window,
+    bubbles: true
+  }), []);
+
+  // Generate an enter press to submit the data on all blurs
+  useEffect(() => {
+
+    if (!forceRefreshValue) return;
+
+    dateInputRef.current.dispatchEvent(enterKeyboardEvent);
+    setIsInputDirty(false);
+    setForceRefreshValue(false);
+
+  }, [ enterKeyboardEvent, forceRefreshValue ]);
+
+
+  useEffect(() => {
+
+    if (disabled) {
+      setFlatpickrInstance(null);
+      return;
+    }
+
+    let config = {
+      allowInput: true,
+      dateFormat: 'm/d/Y',
+      static: true,
+      clickOpens: false
+    };
+
+    if (disallowPassedDates) {
+      config = { ...config, minDate: 'today' };
+    }
+
+    const instance = flatpickr(dateInputRef.current, config);
+
+    setFlatpickrInstance(instance);
+
+    const onCalendarFocusOut = (e) => {
+      if (!instance.calendarContainer.contains(e.relatedTarget)) {
+        instance.close();
+      }
+    };
+
+    // Remove dirty tag to have mouse day selection prioritize input blur
+    const onCalendarMouseDown = (e) => {
+      if (e.target.classList.contains('flatpickr-day')) {
+        setIsInputDirty(false);
+      }
+    };
+
+    // TODO: MORE DETAIL FOR COMMENTS
+    instance.config.onOpen = [
+      () => instance.calendarContainer.addEventListener('focusout', onCalendarFocusOut),
+      () => instance.calendarContainer.addEventListener('mousedown', onCalendarMouseDown),
+    ];
+
+    instance.config.onClose = [
+      () => instance.calendarContainer.removeEventListener('focusout', onCalendarFocusOut),
+      () => instance.calendarContainer.removeEventListener('mousedown', onCalendarMouseDown),
+      () => { setForceRefreshValue(true); }
+    ];
+
+  }, [ disabled, disallowPassedDates, enterKeyboardEvent ]);
+
+  // Shorts the date value back to the source
+  useEffect(() => {
+
+    if (!flatpickrInstance || !flatpickrInstance.config) return;
+
+    flatpickrInstance.setDate(date, true);
+
+  }, [ flatpickrInstance, date.toString() ]);
+
+  // OnChange is updated dynamically, so not to re-render the flatpicker every time it changes
+  useEffect(() => {
+
+    if (!flatpickrInstance || !flatpickrInstance.config) return;
+
+    flatpickrInstance.config.onChange = [ (date) => setDate(new Date(date)) ];
+
+  }, [ flatpickrInstance, setDate ]);
+
+  const focusRelevantCalendarDay = useCallback(
+    (e) => {
+
+      if (!flatpickrInstance) return;
+
+      const container = flatpickrInstance.calendarContainer;
+      const dayToFocus =
+        container.querySelector('.flatpickr-day.selected') ||
+        container.querySelector('.flatpickr-day.today') ||
+        container.querySelector('.flatpickr-day');
+
+      // @ts-ignore
```

Comment:
What happens if we do not `@ts-ignore`?

Reply:
Apparently nothing 🤡 

I'm not certain what changed since then but it used to complain that dayToFocus was not necessarily a focusable HTML element, which broke types.